pilot/pkg/model/sidecar.go

```diff
@@ -217,7 +217,53 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *Config, configNamespa
 	// Assign namespace dependencies
 	out.namespaceDependencies = make(map[string]struct{})
 	for _, listener := range out.EgressListeners {
-		for _, s := range listener.services {
+		// Explicitly callable services are potential destinations, copy those first
+		serviceDestinations := append([]*Service(nil), listener.services...)
```

Comment:
In theory this could be a shallow copy as we only need to append to the slice. Can add this optimization if desired - this felt a bit safer

Reply:
hmm, i agree append can add to the backing array in place, but i don't think that it updates the length of the slice in place. i thought it returns a new slice with a longer length but the same array

Reply:
a more accurate answer is, if you shallow copy a slice, you copy its length and cap. So you can then append it and its guaranteed that you won't affect length and cap of the thing you copied